administrator/components/com_associations/controllers/association.php

```diff
@@ -27,6 +27,20 @@ class AssociationsControllerAssociation extends JControllerForm
 	 */
 	public function cancel($key = null)
 	{
+		$component = $this->input->get('acomponent', '', 'string');
+		$view      = $this->input->get('aview', '', 'string');
+		$extension = $this->input->get('extension', '', 'string');
+		$refID     = $this->input->get('id', '', 'int');
+
+		$getCP = $extension != '' ? ('com_categories.category|' . $extension) : ($component . '.' . $view);
+
+		$checkOutComponent = AssociationsHelper::getComponentProperties($getCP);
+
+		if (!is_null($checkOutComponent->fields->checked_out))
+		{
+			$checkOutComponent->table->checkin(4);
```

Comment:
what is this harcoded "4" ??

also you need to make two checkin, one for reference, one for target.
in other words you need to do a javascript like cancel(array(idreference, idtarget)); (js equivalent)
then process that `$key` in cancel method to checkin both.